src/main/resources/org/sonar/plugins/checkstyle/metadata/additional-rules-meta.yml

```diff
@@ -0,0 +1,10 @@
+rules:
+    - rule: EmptyBlockCheck
+      priority: MAJOR
+#      tag:
+      status: READY
+#      cardinality: # should we keep it here? It's not loaded into repository rule.
```

Comment:
We need to auto figure out this, if we have configurable properties, so we can have few instances of Check.

Reply:
please detect it automatically. it's not additional-rules to be specified by the maintainer but should be detected whenloading and converting metadata

Reply:
How will we get this from checks/meta? For example `tag` is plugin specific and always set by the maintainer I guess, during each check addition?

Reply:
"Ready" mean that Check has no ways to be customized in sonar Ui so it is ready to use by activation.

If Check in configurable, it become a template like, so user needs configure it, give it a name(instance name) and only after activate by name of instance.

Reply:
I didn't get it, every check has default values and don't need compulsory reconfiguring right? Can you please reframe the question?

Reply:
In sonar Ready does not mean ready to use in default form. It means that single instance of Check is enough for all and no other instances are possible/required. If Check is possible to be configurable by user, it is not ready, and user need to make Rule from it first and then activate in some profile.
In your local sonar just try to activate NoClone and AbbreviationNameCheck and will see that workflow of activation is different

Reply:
for cardinality this is the new way to define it: http://javadocs.sonarsource.org/7.9/apidocs/org/sonar/api/rules/Rule.html#setIsTemplate(boolean)
(hence, if we have cardinality multiple, we need to define it as a template)

for status, i'd keep all to READY: http://javadocs.sonarsource.org/7.9/apidocs/org/sonar/api/rules/Rule.html#STATUS_BETA
it does not make sense for us to mark some as BETA or DEPRECATED, @romani ?
whatever checks we have they should all be ready and good to go unless the main project actually marks checks as deprecated, then we can add a lifetime to SQ as well: the check is always READY unless the main repo has a Deprecated annotation present, then it's DEPRECATED and will be removed at some point automatically anyways.

Reply:
@muhlba91 @romani @rnveach To determine cardinality I think we should follow the rules of https://github.com/checkstyle/sonar-checkstyle/issues/87#issue-214878127
Also, I think making multiple instances of a check makes sense when there are different targets to work on (token set/file extensions) right?
So, the condition to have multiple cardinality should be that the check must have either tokens/fileExtension/(or anything I am mising out), and then in addition to these some other configurable properties.

Reply:
After discussing with @romani it has been decided that
1. Cardinality will be always single if there are no configurable property and multiple otherwise in all cases
2. Priority will be MINOR by default
3. Status will be READY by default
4. Tag will have the check package name as tag(imports, headers, block, etc.)

So, only tag will be present as an additional detail YML config, if the user wants to specify some other tag.(@muhlba91 Is it possible for users to specify tags in sonarUI? Then we should remove it from YML, and there will be no need to maintain the YML, hence achieving full automation).

Reply:
tags should be taken from package name of this level https://github.com/checkstyle/checkstyle/tree/master/src/main/java/com/puppycrawl/tools/checkstyle/checks

if module is on this level, then we do no tag. You will see that some Checks at placed on this level for some historical reason.
It will work only for standard Checks, that is still good. But it will not work for thirparty Checks (unpredictable package anmes). So they might need to provide tags to be placed in same tags as standard Checks. Please look at sevntu Checks.
So we need support of yml but for standard Checks we can omit it, or we can provide tags for Checks that are in root package
TodoComment - https://checkstyle.org/config_misc.html#TodoComment , so we can place "misc" tag on it.
https://checkstyle.org/config_misc.html#UncommentedMain . looks like almost all Checks in this root package should have "misc" tag, it might be good to match website grouping.
So lets support this yml file and place there some overrides ... for example Indentation. it is special java package, but group in website is "misc", so lets put tag "misc" on it. All other standard Check in root package could have "misc" tag automatically. So we will have yml in plugin and there will be only "Indentation".
All thirdparty checks (sevntu, ....)  should not fall in "misc" tag if no tags provided.

Reply:
> and there will be only "Indentation".

I didn't understand. When specifying the IndentationCheck in yml, there will be misc tag in the config. What else are you trying to imply?

> All thirdparty checks (sevntu, ....) should not fall in "misc" tag if no tags provided.

It should fall in misc category by default right?

Reply:
>  What else are you trying to imply?

nothing else. in Config we will do overrides of what you determine in from package name. By package name IndentationCheck should has "indentation" tag, but by means of this yml file it will be "misc".

> It should fall in misc category by default right?

no. If no details(no yml file) provided by extension - no tags at all. If extension would like to join standard tags it should provide mapping of Check to them in yml file or old-style config. If extension would like special tags - yml file is required.
Tags are not critical item in sonar. It is easy to find Check even without tags.

Reply:
Okay, done.